projects/storefrontlib/src/cms-components/configurator/commons/config-attribute-types/config-attribute-drop-down/config-attribute-drop-down.component.ts

```diff
@@ -5,29 +5,71 @@ import {
   Input,
   OnInit,
   Output,
+  OnDestroy,
 } from '@angular/core';
+import { formatNumber } from '@angular/common';
 import { FormControl } from '@angular/forms';
-import { Configurator } from '@spartacus/core';
+import { Configurator, LanguageService } from '@spartacus/core';
 import { ConfigFormUpdateEvent } from '../../config-form/config-form.event';
 import { ConfigUIKeyGeneratorService } from '../../service/config-ui-key-generator.service';
+import { Subscription } from 'rxjs';
+import { ConfigUtilsLocalizationService } from '../../service/config-localization-utils.service';
+
 @Component({
   selector: 'cx-config-attribute-drop-down',
   templateUrl: './config-attribute-drop-down.component.html',
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class ConfigAttributeDropDownComponent implements OnInit {
+export class ConfigAttributeDropDownComponent implements OnInit, OnDestroy {
   attributeDropDownForm = new FormControl('');
+  subscriptions = new Subscription();
+  locale: string;
 
-  constructor(public uiKeyGenerator: ConfigUIKeyGeneratorService) {}
+  constructor(
+    public uiKeyGenerator: ConfigUIKeyGeneratorService,
+    public languageService: LanguageService,
+    public utilsService: ConfigUtilsLocalizationService
+  ) {}
 
   @Input() attribute: Configurator.Attribute;
   @Input() group: string;
   @Input() ownerKey: string;
 
+  internalAttribute: Configurator.Attribute;
+
   @Output() selectionChange = new EventEmitter<ConfigFormUpdateEvent>();
 
   ngOnInit() {
     this.attributeDropDownForm.setValue(this.attribute.selectedSingleValue);
+
+    this.subscriptions.add(
+      this.languageService.getActive().subscribe((language) => {
+        this.locale = this.utilsService.getLanguage(language);
+        if (this.attribute.isNumeric) {
+          this.internalAttribute = { ...this.attribute, values: [] };
+
+          this.attribute.values.forEach((element) => {
+            let internalNumber: string;
+
+            internalNumber = formatNumber(
+              Number(
+                element.valueDisplay
```

Comment:
Why is this needed? For numeric input at least the value was available in en locale, let's discuss.